.travis-ci.sh

```diff
@@ -53,26 +52,33 @@ critical errors that must be understood and fixed before your pull
 request can be merged.
 ------------------------------------------------------------------------
 EOF
-    mkdir -p $PREFIX
+  case $XARCH in
+  x64)
     ./configure --prefix $PREFIX -with-debug-runtime \
       -with-instrumented-runtime $CONFIG_ARG
-    export PATH=$PREFIX/bin:$PATH
-    $MAKE world.opt
-    $MAKE ocamlnat
-    (cd testsuite && $MAKE all)
-    (cd testsuite && $MAKE USE_RUNTIME="d" all)
-    $MAKE install
-    # check_all_arches checks tries to compile all backends in place,
-    # we need to redo (small parts of) world.opt afterwards
-    $MAKE check_all_arches
-    $MAKE world.opt
-    $MAKE manual-pregen
+    ;;
+  i386)
+    ./configure --prefix $PREFIX -with-debug-runtime \
+      -with-instrumented-runtime $CONFIG_ARG \
+      -host i686-pc-linux-gnu
     ;;
   *)
     echo unknown arch
     exit 1
     ;;
   esac
+
+  export PATH=$PREFIX/bin:$PATH
+  $MAKE world.opt
+  $MAKE ocamlnat
+  (cd testsuite && $MAKE all)
+  [ $XARCH =  "i386" ] ||  (cd testsuite && $MAKE USE_RUNTIME="d" all)
```

Comment:
I think this would be clearer with a `case` statement.

Reply:
This is a temporary change, I was going for brevity, rather than clarity!

Reply:
(temporary in the sense that I hope @damiendoligez may be able to advise what's wrong with the freelist accounting in the coming weeks)